cpp/src/join/mixed_join_common_utils.cuh

```diff
@@ -160,6 +161,42 @@ struct pair_expression_equality : public expression_equality<has_nulls> {
   }
 };
 
+/**
+ * @brief Equality comparator that composes two row_equality comparators.
+ */
+class double_row_equality {
+ public:
+  double_row_equality(row_equality equality_comparator, row_equality conditional_comparator)
+    : _equality_comparator{equality_comparator}, _conditional_comparator{conditional_comparator}
+  {
+  }
+
+  __device__ bool operator()(size_type lhs_row_index, size_type rhs_row_index) const noexcept
+  {
+    using experimental::row::lhs_index_type;
+    using experimental::row::rhs_index_type;
+
+    return _equality_comparator(lhs_index_type{lhs_row_index}, rhs_index_type{rhs_row_index}) &&
+           _conditional_comparator(lhs_index_type{lhs_row_index}, rhs_index_type{rhs_row_index});
+  }
+
+ private:
+  row_equality _equality_comparator;
+  row_equality _conditional_comparator;
+};
+
+// The hash table type used by mixed_semi_join with the build_table.
+using hash_set_type = cuco::static_set<hash_value_type,
+                                       cuco::extent<size_t>,
+                                       cuda::thread_scope_device,
+                                       double_row_equality,
+                                       cuco::linear_probing<1, row_hash>,
```

Comment:
Have you tuned CG size like 2 or 4 as well as the probing scheme like double hashing?

Reply:
I tried using CG 2, 4. And CG with 4 gave closer performance to current implementation with static_map. 
**double hashing** is not supported with **static_set** `.with_hash_function` API yet.

Reply:
@srinivasyadav18 by closer performance, do you mean closer but slower or closer or faster? What about cg_size=1? I can probably benchmark and tune cg_size again but would help if you have any data.

Reply:
Update: I tried with cg_size=1,2,4 and 4 seems to be the best among them but only marginally so going with that.